epoxy-processor/src/main/java/com/airbnb/epoxy/EpoxyProcessor.java

```diff
@@ -662,8 +662,8 @@ private MethodSpec generateEquals(ClassToGenerateInfo helperClass) {
               .endControlFlow();
         }
       } else {
-        builder.beginControlFlow("if ($L != null && that.$L == null"
-                + " || $L == null && that.$L != null)",
+        builder.beginControlFlow("if (($L != null && that.$L == null)"
```

Comment:
It is a simply XOR operation. I didn't quite catch that. Thanks for pointing out :) I will make the change and update it.